### PR TITLE
REF/TST: method-specific files for combine, update; parametrize

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/tests/series/test_combine_concat.py [hunk 1]
```diff
@@ -2,84 +2,27 @@
...
-            # other is object, cannot be cast
-            ([(61,), (63,)], "int32", pd.Series([10, (61,), 12])),
-            ([(61,), (63,)], "int64", pd.Series([10, (61,), 12])),
-            ([(61,), (63,)], float, pd.Series([10.0, (61,), 12.0])),
-            ([(61,), (63,)], object, pd.Series([10, (61,), 12])),
-        ],
+        "dtype", ["float64", "int8", "uint8", "bool", "m8[ns]", "M8[ns]"]
     )
-    def test_update_dtypes(self, other, dtype, expected):
+    def test_concat_empty_series_dtypes_match_roundtrips(self, dtype):
```
Comment (simonjayhawkins) [inline pandas/tests/series/test_combine_concat.py]: this change looks good but better if not done at the same time as a move (in future).  could any of the dtype fixtures in conftest be used here instead? (as a follow-up)

File: pandas/tests/series/test_combine_concat.py [hunk 2]
```diff
@@ -118,61 +61,45 @@ def get_result_type(dtype, dtype2):
...
+            # datetime-like
+            ("m8[ns]", np.bool, np.object_),
+            ("m8[ns]", np.int64, np.object_),
+            ("M8[ns]", np.bool, np.object_),
+            ("M8[ns]", np.int64, np.object_),
+            # categorical
+            ("category", "category", "category"),
+            ("category", "object", "object"),
+        ],
+    )
```
Comment (simonjayhawkins) [inline pandas/tests/series/test_combine_concat.py]: again keep test refactoring separate from moving things. (in future)

**Review by simonjayhawkins — APPROVED** [review]
@jbrockmendel lgtm

File: pandas/tests/series/test_combine_concat.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/series/test_combine_concat.py]: good reminder

**Comment by simonjayhawkins** [comment]
Thanks @jbrockmendel

**Pull request merged**